src/UI/Internal/FiltersHeaders.elm

```diff
@@ -287,6 +308,180 @@ singleTextFilterRender renderConfig applyMsg { fromFiltersMsg, index, label } ed
     editable
         |> Filters.editableWithDefault ""
         |> TextField.singlelineText editMsg label
+        |> TextField.withSize Size.extraSmall
         |> TextField.withWidth TextField.widthFull
         |> TextField.withOnEnterPressed applyMsg
         |> TextField.renderElement renderConfig
+
+
+multiTextFilterRender :
+    RenderConfig
+    -> msg
+    -> Config msg
+    -> Filters.Editable (Array String)
+    -> Element msg
+multiTextFilterRender renderConfig applyMsg { fromFiltersMsg, index, label } editableArr =
+    let
+        editMsg subIndex str =
+            fromFiltersMsg <| Filters.EditMultiText { column = index, field = subIndex, value = str }
+
+        single subIndex line =
+            line
+                |> TextField.singlelineText (editMsg subIndex) label
+                |> TextField.withSize Size.extraSmall
+                |> TextField.withWidth TextField.widthFull
+                |> TextField.withOnEnterPressed applyMsg
+                |> TextField.renderElement renderConfig
+
+        instanceArr =
+            Filters.editableWithDefault Array.empty editableArr
+    in
+    instanceArr
+        |> Array.push ""
+        |> Array.indexedMap single
+        |> Array.toList
+        |> Element.column [ Element.width fill, Element.spacing 8 ]
+
+
+selectFilterRender :
+    RenderConfig
+    -> Config msg
+    -> List String
+    -> Filters.Editable Int
+    -> Element msg
+selectFilterRender renderConfig { fromFiltersMsg, index } list { current, applied } =
+    let
+        editMsg subIndex =
+            fromFiltersMsg <| Filters.EditSelect { column = index, value = subIndex }
+
+        selected =
+            maybeNotThen applied current
+    in
+    radioGroup renderConfig
```

Comment:
I would use `radioGroup` qualified, to be more explicit that's an external funcion.